pkg/kubefedctl/federate/util.go

```diff
@@ -207,7 +207,11 @@ func getResourcesInNamespace(config *rest.Config, namespace string, skipAPIResou
 		targetResources := resources{apiResource: apiResource}
 		for _, item := range resourceList.Items {
 			resource := item
-			targetResources.resources = append(targetResources.resources, &resource)
+			if len(validation.IsDNS1123Subdomain(resource.GetName())) == 0 {
+				targetResources.resources = append(targetResources.resources, &resource)
+			} else {
+				klog.Warningf("Skipping resource %s of type %s", resource.GetName(), apiResource.Name)
```

Comment:
We should add the reason it's being skipped i.e. does not conform to DNS 1123 spec. Additionally, I think it would be good to spit out the errors from the call to `IsDNS1123Subdomain` so that users are aware of what exactly does not conform.